src/lib/core/a11y/interactivity-checker.ts

```diff
@@ -30,26 +33,93 @@ export class InteractivityChecker {
    * being clipped by an `overflow: hidden` parent or being outside the viewport.
    */
   isVisible(element: HTMLElement) {
-    // There are additional special cases that this does not capture, but this will work for
-    // the most common cases.
+    let nodeName = element.nodeName.toLowerCase();
+    let isControlAudio = nodeName === 'audio' && (element as HTMLAudioElement).controls;
 
-    // Use logic from jQuery to check for `display: none`.
-    // See https://github.com/jquery/jquery/blob/master/src/css/hiddenVisibleSelectors.js#L12
-    if (!(element.offsetWidth || element.offsetHeight || element.getClientRects().length)) {
+    // In IE11 audio elements with controls have invalid rectangles, but are still visible.
+    if (!isControlAudio && !checkRectangles()) {
       return false;
     }
 
-    // Check for css `visibility` property.
-    // TODO(jelbourn): do any browsers we support return an empty string instead of 'visible'?
-    return getComputedStyle(element).getPropertyValue('visibility') == 'visible';
+    return getComputedStyle(element).getPropertyValue('visibility') === 'visible';
+
+    function checkRectangles() {
+      // Use logic from jQuery to check for an invisible element.
+      // See https://github.com/jquery/jquery/blob/master/src/css/hiddenVisibleSelectors.js#L12
+      return element.offsetWidth || element.offsetHeight || element.getClientRects().length;
+    }
   }
 
   /**
    * Gets whether an element can be reached via Tab key.
    * Assumes that the element has already been checked with isFocusable.
    */
   isTabbable(element: HTMLElement) {
-    // Again, naive approach that does not capture many special cases and browser quirks.
+
+    let nodeName = element.nodeName.toLowerCase();
+    let frameElement = getWindow(element).frameElement as HTMLElement;
+    let frameType = frameElement && frameElement.nodeName.toLowerCase();
+    let tabIndexValue = getTabIndexValue(element);
+
+    if (frameElement) {
+
+      // Frame elements inherit their tabindex onto all child elements.
+      if (getTabIndexValue(frameElement) === -1) {
```

Comment:
so if an element inherits its tabindex through the frame it's definitely disabled? (because you have a few places below where if the user sets tabindex=-1 that doesn't necessarily mean non-tabbable, e.g. blink audio)